consensus/consensus-types/src/wrapped_ledger_info.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::vote_data::VoteData;
+use anyhow::{ensure, Context};
+use aptos_types::{
+    aggregate_signature::AggregateSignature,
+    block_info::BlockInfo,
+    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+    validator_verifier::ValidatorVerifier,
+};
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+/// This struct is similar to QuorumCert, except that the verify function doesn't verify vote_data.
+/// vote_data and consensus_data_hash inside signed_ledger_info could be replaced with dummy values.
+#[derive(Deserialize, Serialize, Clone, Debug, Eq, PartialEq)]
+pub struct WrappedLedgerInfo {
+    /// The vote information certified by the quorum.
+    vote_data: VoteData,
+    /// The signed LedgerInfo of a committed block that carries the data about the certified block.
+    signed_ledger_info: LedgerInfoWithSignatures,
+}
+
+impl Display for WrappedLedgerInfo {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "WrappedLedgerInfo: [{}, {}]",
+            self.vote_data, self.signed_ledger_info
+        )
+    }
+}
+
+impl WrappedLedgerInfo {
+    pub fn new(vote_data: VoteData, signed_ledger_info: LedgerInfoWithSignatures) -> Self {
+        WrappedLedgerInfo {
+            vote_data,
+            signed_ledger_info,
+        }
+    }
+
+    #[cfg(any(test, feature = "fuzzing"))]
+    pub fn dummy() -> Self {
+        Self {
+            vote_data: VoteData::dummy(),
+            signed_ledger_info: LedgerInfoWithSignatures::new(
+                LedgerInfo::dummy(),
+                AggregateSignature::empty(),
+            ),
+        }
+    }
+
+    pub fn certified_block(&self) -> &BlockInfo {
+        self.vote_data.proposed()
+    }
+
+    pub fn parent_block(&self) -> &BlockInfo {
```

Comment:
ditto

Reply:
fast_forward_sync in sync_manager is still utilizing this information when order_vote_enabled flag is set to false.

Reply:
Removed it.